### PR TITLE
Implement types.Blob.Concat and types.NewBlobP

### PR DESCRIPTION
**Pull request by kalman**
Concat is a simple use of the sequence concat code that List.Concat
uses. NewBlobP uses Concat to construct a Blob in parallel.

Perf tests for NewBlobP write N temporary files then constructs a Blob
from them, so there is some I/O, but it appears to be mostly CPU bound.
NewBlobP doesn't get much more than 50% faster with any P >= 2.

### CODE REVIEW

**Comment by kalman** [comment]
@willhite or @rafael-atticlabs PTAL

File: go/types/list_test.go [hunk 1]
```diff
@@ -1107,3 +1107,17 @@ func TestListConcat(t *testing.T) {
 	run(6, 1e4, 0, 1000, 10)
 	run(7, 1e4, 1e4-1000, 1e4, 10)
 }
+
+func TestListConcatDifferentTypes(t *testing.T) {
```
Comment (kalman) [inline go/types/list_test.go]: Not related to this patch in any way, it just occurred to me to test it.

File: go/types/blob.go [hunk 2]
```diff
@@ -172,10 +186,46 @@ func chunkBlobLeaf(vr ValueReader, buff []byte) (Collection, orderedKey, uint64)
 	return blob, orderedKeyFromInt(len(buff)), uint64(len(buff))
 }
 
+// NewBlob creates a Blob by reading from r.
 func NewBlob(r io.Reader) Blob {
 	return NewStreamingBlob(r, nil)
 }
 
+// NewBlobP creates a Blob by reading from every Reader in rs in parallel, then
```
Comment (kalman) [inline go/types/blob.go]: I'm considering just making NewBlob/NewStreamingBlob do the parallelisation automatically, and not having NewBlobP. WDYT?

File: go/types/list_test.go (hunk 1 above)
Comment (rafael-atticlabs) [inline go/types/list_test.go]: nice

**Comment by rafael-atticlabs** [comment]
very cool. lgtm.

File: go/types/blob.go (hunk 2 above)
Comment (rafael-atticlabs) [inline go/types/blob.go]: makes sense to me.

Comment (kalman) [inline go/types/blob.go]: done

**Pull request merged**